server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1301,10 +1304,11 @@ public boolean isReadOnly() {
      * @param repositoryData RepositoryData to write
      * @param expectedGen    expected repository generation at the start of the operation
      * @param writeShardGens whether to write {@link ShardGenerations} to the new {@link RepositoryData} blob
+     * @param stateFilter    filter for the last cluster state update executed by this method
```

Comment:
It's not a filter, rather a transformation function. Perhaps adapt naming?

Reply:
Renamed :)